### PR TITLE
Change Android package and class names away from `mapbox`

### PR DESCRIPTION
**Pull request by fynngodau**
* [x] Change package name to `org.maplibre.android`
* [x] Change class name of `Mapbox` and `MapboxMap` to `MapLibre` and `MapLibreMap`
* [x] Append to changelog

Fixes https://github.com/maplibre/maplibre-native/issues/1155.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/cpp/jni_native.cpp [hunk 1]
```diff
@@ -162,7 +162,7 @@ void registerNatives(JavaVM* vm) {
     Logger::registerNative(env);
 
     // AssetManager
-    Mapbox::registerNative(env);
+    Maplibre::registerNative(env);
```
Comment (birkskyum) [inline platform/android/MapboxGLAndroidSDK/src/cpp/jni_native.cpp:165]: can we have MapLibre with capital L?

**Comment by fynngodau** [comment]
@birkskyum See also the discussion on Slack: https://osmus.slack.com/archives/C04G140P9U6/p1685131968555689?thread_ts=1685104345.245239&cid=C04G140P9U6

**Comment by louwers** [comment]
@birkskyum So the idea is to apply camel casing consistently. It makes it a bit easier to read identifiers e.g. compare 

- `MapLibreMap` vs `MaplibreMap`
- `MapLibreConfigurationException` vs `MaplibreConfigurationException`

maplibre-gl-js  has e.g. `MapLibreEvent`, `MapLibreZoomEvent` but also `MaplibreWorker`

What do you think? I can see a benefit of sticking to the consistent brand identity as well. It is pretty easy to change now before this is merged.

**Comment by birkskyum** [comment]
> @birkskyum So the idea is to apply camel casing consistently. It makes it a bit easier to read identifiers e.g. compare
> 
> * `MapLibreMap` vs `MaplibreMap`
> * `MapLibreConfigurationException` vs `MaplibreConfigurationException`
> 
> maplibre-gl-js has e.g. `MapLibreEvent`, `MapLibreZoomEvent` but also `MaplibreWorker`
> 
> What do you think? I can see a benefit of sticking to the consistent brand identity as well. It is pretty easy to change now before this is merged.

I'm in favor of consistency, so I'd suggest to stick with `MapLibre` as it's together with `maplibre` what's used virtually everywhere else, and we can easily update the rest to this convention if there are outliers. If the entire organization decided to change the name of to Maplibre, then I'd be in favor of updating the spelling _**everywhere**_ to that, to stay consistent, but I have never seen anyone even mention such a change. Another thing is that this codebase already made a strong buy-in to the capital L when changing the namespace to MLN <=> MapLibre Native, which would be partially undermined by lowercasing the L in the SDKs.

**Review by louwers — COMMENTED** [review]
The instrumentation tests are not running anymore. I am having a look.

File: platform/android/MapboxGLAndroidSDKTestApp/build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ android {
     compileSdkVersion androidVersions.compileSdkVersion
 
     defaultConfig {
-        applicationId "com.mapbox.mapboxsdk.testapp"
+        applicationId "org.maplibre.testapp"
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDKTestApp/build.gradle:12]: @fynngodau Is this correct?

**Comment by louwers** [comment]
This needs to be updated

https://github.com/maplibre/maplibre-native/blob/f9cefa55a68ae51c5d9f3a7d69f4e31a4e089ce9/platform/android/MapboxGLAndroidSDKTestApp/build.gradle#L17

File: platform/android/MapboxGLAndroidSDKTestApp/build.gradle (hunk 2 above)
Comment (fynngodau) [reply platform/android/MapboxGLAndroidSDKTestApp/build.gradle:12]: Yes, the package name can be arbitrary (we talked about this on Slackl.

Comment (louwers) [reply platform/android/MapboxGLAndroidSDKTestApp/build.gradle:12]: Ah I remember.

**Comment by fynngodau** [comment]
@louwers I think you had forgotten to rename the class files when renaming the classes in my branch. I have pushed a fix. Additionally, I have pushed a commit that fixes the line you showed. Instrumentation tests now seem to run again.

**Comment by louwers** [comment]
@fynngodau Cool, thank you. I will take another look and then merge.

**Comment by louwers** [comment]
Are the instrumentation tests running through for you? They run now, but I am getting "MapboxMap isn't initialised" in `BaseTest.java`.

**Review by louwers — APPROVED** [review]
Thanks for this! :slightly_smiling_face:

**Pull request merged**